src/Datadog.Trace/Sampling/DefaultSamplingRule.cs

```diff
@@ -8,7 +8,7 @@ internal class DefaultSamplingRule : ISamplingRule
     {
         private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<DefaultSamplingRule>();
 
-        private static Dictionary<SampleRateKey, float> _sampleRates = new Dictionary<SampleRateKey, float>();
+        private Dictionary<SampleRateKey, float> _sampleRates = new Dictionary<SampleRateKey, float>();
```

Comment:
This static field was set through an instance method, giving the wrong expectation and probably causing the flaky sampler test results.

Reply:
I can't believe I scanned the code so many times and missed this